### PR TITLE
Modify errorcode cannot be exported and cannot be imported correctly - CR changes for 1469

### PR DESCRIPTION
**Pull request by martin-grofcik**
https://github.com/flowable/flowable-engine/pull/1469

### CODE REVIEW

**Comment by martin-grofcik** [comment]
Error definition field on the process level
image

Error definition dialog
image

Error reference
image

```
<definitions xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:flowable="http://flowable.org/bpmn" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:omgdc="http://www.omg.org/spec/DD/20100524/DC" xmlns:omgdi="http://www.omg.org/spec/DD/20100524/DI" typeLanguage="http://www.w3.org/2001/XMLSchema" expressionLanguage="http://www.w3.org/1999/XPath" targetNamespace="http://www.flowable.org/processdef">
  <error id="myerrorId1" errorCode="myerrorCode1"></error>
  <error id="myerrorId2" errorCode="myerrorCode2"></error>
  <error id="myerrorId3" errorCode="myerrorCode3"></error>
  <error id="myerrorFreeText" errorCode="myerrorFreeText"></error>
  <process id="errorProcess3.2" name="errorProcess3.2" isExecutable="true">
    <startEvent id="sid-8895B7A0-F8E5-42C3-B292-87C215C47B6E" isInterrupting="true">
      <errorEventDefinition errorRef="myerrorFreeText"></errorEventDefinition>
    </startEvent>

```

File: modules/flowable-bpmn-model/src/main/java/org/flowable/bpmn/model/BpmnModel.java [hunk 1]
```diff
@@ -423,8 +423,12 @@ public void setErrors(Map<String, String> errorMap) {
     }
 
     public void addError(String errorRef, String errorCode) {
-        if (StringUtils.isNotEmpty(errorRef)) {
-            errorMap.put(errorRef, errorCode);
+        if (StringUtils.isNotEmpty(errorRef) && !errorMap.containsKey(errorRef)) {
```
Comment (martin-grofcik) [inline modules/flowable-bpmn-model/src/main/java/org/flowable/bpmn/model/BpmnModel.java]: What I am not sure about is whether we can merge errors in this way.

Comment (qiudaoke) [reply modules/flowable-bpmn-model/src/main/java/org/flowable/bpmn/model/BpmnModel.java]: @martin-grofcik  I think it is correct

**Comment by PascalSchumacher** [comment]
👍  Imho it would be really great if this feature was merged.

**Comment by tijsrademakers** [comment]
Closing this PR for now. The problem with the PR is the possible issues with backward compatibility, how would it work for existing models with error events? Please reopen if you think this should still be added.

**Pull request closed without merging**